bus-mapping/src/circuit_input_builder.rs

```diff
@@ -809,6 +808,210 @@ impl<'a> CircuitInputStateRef<'a> {
     }
 }
 
+/// State and Code Access with "keys/index" used in the access operation.
+#[derive(Debug, PartialEq)]
+enum AccessValue {
+    Account { address: Address },
+    Storage { address: Address, key: Word },
+    Code { address: Address },
+}
+
+/// State Access caused by a transaction or an execution step
+#[derive(Debug, PartialEq)]
+struct Access {
+    step_index: Option<usize>,
+    rw: RW,
```

Comment:
Curious about what's the purpose of `rw` when `Access` is used to query the account details from geth, do we need to call RPC in different way for read and write?

Reply:
I'm unsure that this is used to do any queries to Geth. But it's indeed the result of the query layed down in a struct.

So that with it we can build the `AccessTrace`.

Is it the purpose @ed255 ?

Reply:
You're right to be curious because the `rw` is indeed not used here.  The reason I put it here was because it's information that is easy to gather, I think it helps when debugging (it's easier to match an access with the step that caused it if it also records the rw) and also I had the intuition that in the future it may be useful:
- If we also record if the write access is permanent or not we can reduce the amount of geth queries:
    - if the first time in a tx a code is accessed is for write and it's persistent, it means successful code creation; so there's no need to query that code from geth, because it didn't exist before.
Does this seam reasonable?  If you think having the `rw` here (which is not used at all now) is confusing, I can remove it.

Reply:
It makes sense to me now, there might be some cases that we don't need to query geth just like you said.